.github/workflows/process_new_pr.yml

```diff
@@ -0,0 +1,127 @@
+name: Assign Reviewers for Contributors
+
+on:
+  pull_request_target:
+    types: [opened]
+
+permissions: {}
+
+jobs:
+  check_contributor_and_reviewers:
+    name: Check Contributor Status and Assigned Reviewers
+    runs-on: ubuntu-22.04
+    permissions:
+      pull-requests: read
+    outputs:
+      is_new_contributor: ${{ steps.check-new-contributor.outputs.result }}
+      has_reviewers: ${{ steps.check-reviewers.outputs.has_reviewers }}
+      has_teams: ${{ steps.check-reviewers.outputs.has_teams }}
+      author: ${{ steps.set-env.outputs.author }}
+      repo: ${{ steps.set-env.outputs.repo }}
+      owner: ${{ steps.set-env.outputs.owner }}
+      pr_number: ${{ steps.set-env.outputs.pr_number }}
+    steps:
+      - name: Extract PR Metadata
+        id: set-env
+        shell: bash
+        run: |
+          echo "author=${{ github.event.pull_request.user.login }}" >> $GITHUB_OUTPUT
+          echo "repo=${{ github.event.repository.name }}" >> $GITHUB_OUTPUT
+          echo "owner=${{ github.event.repository.owner.login }}" >> $GITHUB_OUTPUT
+          echo "pr_number=${{ github.event.pull_request.number }}" >> $GITHUB_OUTPUT
+      - name: Check if PR Author is a New Contributor
+        id: check-new-contributor
+        uses: actions/github-script@v7
+        with:
+          github-token: ${{ secrets.GITHUB_TOKEN }}
+          script: |
+            const author = '${{ github.event.pull_request.user.login }}';
+            const owner = '${{ github.event.repository.owner.login }}';
+            const repo = '${{ github.event.repository.name }}';
+            
+            try {
+              const { data } = await github.rest.search.issuesAndPullRequests({
+                q: `repo:${owner}/${repo} type:pr is:merged author:${author}`,
+              });
+              return data.total_count === 0;
+            } catch (error) {
+              console.error('Error checking contributor status:', error);
+              return false;
+            }
+      - name: Check for Assigned Reviewers and Teams
```

Comment:
this checks for reviews being requested from users/teams, not whether they're assigned

also, what if this workflow runs before oppiabot does the reviewer assignments? seems like there's a race condition here. Or is this supposed to replace oppiabot? If so that needs to be made clear, and we need a plan to disable the replaced functionality in oppiabot concurrent with this PR merging